packages/babel-plugin-check-es2015-constants/src/index.js

```diff
@@ -7,7 +7,53 @@ export default function({ messages }) {
           if (binding.kind !== "const" && binding.kind !== "module") continue;
 
           for (const violation of (binding.constantViolations: Array)) {
-            throw violation.buildCodeFrameError(messages.get("readOnly", name));
+            const throwNode = t.throwStatement(
+              t.newExpression(t.identifier("Error"), [
+                t.stringLiteral(messages.get("readOnly", name)),
+              ]),
+            );
+
+            if (violation.isAssignmentExpression()) {
+              violation
+                .get("right")
+                .replaceWith(
+                  t.sequenceExpression([
+                    t.callExpression(
+                      t.functionExpression(
+                        null,
+                        [],
+                        t.blockStatement([throwNode]),
+                      ),
+                      [],
+                    ),
+                    violation.get("right").node,
+                  ]),
+                );
+            } else if (violation.parentPath.isUpdateExpression()) {
```

Comment:
Currently there is only one in `packages/babel-plugin-check-es2015-constants/test/fixtures/general/update-expression`. Could probably use more since this can show up in a lot of wierd places I think.